src/VecSim/algorithms/svs/svs_batch_iterator.h

```diff
@@ -28,10 +28,8 @@ class SVS_BatchIterator : public VecSimBatchIterator {
     using impl_type = std::invoke_result_t<mkbi_t, Index, query_type, size_t>;
 
     using dist_type = typename Index::distance_type;
-    bool done;
     size_t dim;
-    const Index *index_; // Pointer to the index, used for reset and other operations.
-    std::unique_ptr<impl_type> impl_;
+    impl_type impl_;
```

Comment:
why the change?

Reply:
Because we do not need to 'reconstruct' SVS internal batch iterator implementation anymore during the lifetime of `SVS_BatchIterator`.
So, we have no need to keep a pointer to index, and we do not need to wrap iterator implementation to `unique_ptr`